pkg/gui/presentation/branches.go

```diff
@@ -125,12 +131,12 @@ func getBranchDisplayStrings(
 
 // GetBranchTextStyle branch color
 func GetBranchTextStyle(name string) style.TextStyle {
-	branchType := strings.Split(name, "/")[0]
-
-	if value, ok := branchPrefixColorCache[branchType]; ok {
-		return value
+	if style, ok := branchPrefixColorCache.match(name); ok {
+		return *style
 	}
 
+	// Default colors for common branch types
+	branchType := strings.Split(name, "/")[0]
```

Comment:
keeping the old default behavior

Reply:
@jesseduffield How do you feel about removing this default behavior? I find it surprising and annoying. I don't want it, but it's not obvious how to disable it; previously I had to use this to disable it:
```
gui:
  branchColors:
    feature: default
    bugfix: default
    hotfix: default
```
but if I want to use the new regex patterns now, I have to say
```
gui:
  branchColorPatterns:
    "mypattern": "#ff0000"
    "^feature/": default
    "^bugfix/": default
    "^hotfix/": default
```
This is very non-obvious, and I feel like I shouldn't have to do that.